src/libcore/ops.rs

```diff
@@ -95,6 +95,16 @@ use fmt;
 #[stable(feature = "rust1", since = "1.0.0")]
 pub trait Drop {
     /// A method called when the value goes out of scope.
+    ///
+    /// When this method has been called, `self` has not yet been deallocated.
+    /// If it were, `self` would be a dangling reference.
+    ///
+    /// After this function is over, the memory of `self` will be deallocated.
+    ///
+    /// # Panics
+    ///
+    /// Given that a `panic!` will call `drop()` as it unwinds, any `panic!` in
+    /// a `drop()` implementation will likely abort.
```

Comment:
Here's the [double panic detector](https://github.com/rust-lang/rust/blob/5b838c586c4da31395d614d9f076430314fa05ff/src/libstd/panicking.rs#L229-L237); the comment implies that something besides an abort could be done but it would be tricky. Scary-sounding alternatives like calling `std::thread::panicking` or using `panic::recover` should maybe be in the Nomicon instead of spelled out here, I would say?